spark/src/test/scala/org/apache/spark/sql/delta/DeltaVacuumSuite.scala

```diff
@@ -578,6 +586,230 @@ class DeltaVacuumSuite
     }
   }
 
+  test("schema validation for vacuum by using inventory dataframe") {
+    withEnvironment { (tempDir, clock) =>
+      val deltaLog = DeltaLog.forTable(spark, tempDir, clock)
+      val txn = deltaLog.startTransaction()
+      val schema = new StructType().add("_underscore_col_", IntegerType).add("n", IntegerType)
+      val metadata =
+        Metadata(schemaString = schema.json, partitionColumns = Seq("_underscore_col_"))
+      txn.commit(metadata :: Nil, DeltaOperations.CreateTable(metadata, isManaged = true))
+
+      val expectedInventorySchema = StructType(
+        Seq(
+          StructField("path", StringType),
+          StructField("length", LongType),
+          StructField("isDir", BooleanType),
+          StructField("modificationTime", LongType)
+        ))
+      val inventorySchema = StructType(
+        Seq(
+          StructField("file", StringType),
+          StructField("size", LongType),
+          StructField("isDir", BooleanType),
+          StructField("modificationTime", LongType)
+        ))
+
+      // Create a Seq of Rows containing the data
+      val data = Seq(
+        Row(s"${deltaLog.dataPath}/file1.txt", 300000L, false, 0L),
+        Row(s"${deltaLog.dataPath}/file2.txt", 300000L, false, 0L),
+        Row(s"${deltaLog.dataPath}/_underscore_col_=10/test.txt", 300000L, false, 0L),
+        Row(s"${deltaLog.dataPath}/_underscore_col_=10/test2.txt", 300000L, false, 0L)
+      )
+
+      val inventory = spark.createDataFrame(spark.sparkContext.parallelize(data), inventorySchema)
+
+      gcTest(deltaLog, clock)(
+        CreateFile("file1.txt", commitToActionLog = true, Map("_underscore_col_" -> "10")),
+        CreateFile("file2.txt", commitToActionLog = false, Map("_underscore_col_" -> "10")),
+        CreateFile("_underscore_col_=10/test.txt", true, Map("_underscore_col_" -> "10")),
+        CreateFile("_underscore_col_=10/test2.txt", false, Map("_underscore_col_" -> "10")),
+        CheckFiles(Seq("file1.txt", "_underscore_col_=10")),
+        CheckFiles(Seq("file2.txt", "_underscore_col_=10")),
+        LogicallyDeleteFile("_underscore_col_=10/test.txt"),
+        AdvanceClock(defaultTombstoneInterval + 1000),
+        ExpectFailure(
```

Comment:
This test can probably do without all the prep work -- it will probably fail just fine on an empty table. :)